db/db.go

```diff
@@ -26,6 +27,9 @@ func SetupDBConnection(db string) (*mgo.Session, *mgo.Database) {
 }
 
 func connectDB(s *mgo.Session, dbName string) *mgo.Database {
+	mux := sync.Mutex{}
+	mux.Lock()
+	defer mux.Unlock()
```

Comment:
This won't work because you defined `mux` inside the function -- so every time the function is called, it will create a new mutex for locking and unlocking (which defeats the point).  This would work better if you defined `mux` outside of the function.

That said, Go has an even better solution for doing a one-time action (which is basically what I think you're trying to do here).  Check out [sync.Once](https://golang.org/pkg/sync/#Once).  I think you'd use it here like this:

```Go
var once sync.Once

func connectDB(s *mgo.Session, dbName string) *mgo.Database {
    once.Do(func() {
        db = s.DB(dbName)
    })
    return db;
}
```

That said, what do you want to happen if someone passes in a _different_ `dbName` than it was originally initialized with?  It seems it should throw some kind of error or something?  Should you check for that?